gmso/external/convert_hoomd.py

```diff
@@ -0,0 +1,881 @@
+        )
+
+    agtype_group_map = {
+        "HarmonicAnglePotential": {
+            "container": hoomd.md.angle.Harmonic,
+            "parser": _parse_harmonic,
+        },
+    }
+    angle_forces = list()
+    for group in groups:
+        angle_forces.append(
+            agtype_group_map[group]["parser"](
+                container=agtype_group_map[group]["container"](),
+                agtypes=groups[group],
+            )
+        )
+
+    return angle_forces
+
+
+def _parse_dihedral_forces(top, potential_types, potential_refs, base_units):
+    """Parse dihedral forces."""
+
+    def _parse_periodic(container, dtypes):
+        for dtype in dtypes:
+            # TODO: Unit conversion
+            container.params["-".join(dtype.member_types)] = {
+                "k": dtype.parameters["k"],
+                "d": 1,
+                "n": dtype.parameters["n"],
+                "phi0": dtype.parameters["phi_eq"],
+            }
+        return container
+
+    def _parse_opls(container, dtypes):
+        for dtype in dtypes:
+            # TODO: Unit conversion
+            # TODO: The range of ks is mismatched (GMSO go from k0 to k5)
+            # May need to do a check that k0 == k5 == 0 or raise a warning
+            container.params[dtype.member_types] = {
+                "k1": dtype.parameters["k1"],
+                "k2": dtype.parameters["k2"],
+                "k3": dtype.parameters["k3"],
+                "k4": dtype.parameters["k4"],
+            }
+        return container
+
+    def _parse_rb(container, dtypes):
+        warnings.warn(
+            "RyckaertBellemansTorsionPotential will be converted to OPLSTorsionPotential."
+        )
+        for dtype in dtypes:
+            opls = convert_ryckaert_to_opls(dtype)
+            # TODO: Unit conversion
+            # TODO: The range of ks is mismatched (GMSO go from k0 to k5)
+            # May need to do a check that k0 == k5 == 0 or raise a warning
+            container.params["-".join(dtype.member_types)] = {
+                "k1": opls.parameters["k1"],
+                "k2": opls.parameters["k2"],
+                "k3": opls.parameters["k3"],
+                "k4": opls.parameters["k4"],
+            }
+        return container
+
+    unique_dtypes = top.dihedral_types(
+        filter_by=PotentialFilters.UNIQUE_NAME_CLASS
+    )
+    groups = dict()
+    for dtype in unique_dtypes:
+        group = potential_types[dtype]
+        if group not in groups:
+            groups[group] = [dtype]
+        else:
+            groups[group].append(dtype)
+
+    for group in groups:
+        expected_units_dim = potential_refs[group][
+            "expected_parameters_dimensions"
+        ]
+        groups[group] = _convert_params_units(
+            groups[group], expected_units_dim, base_units
+        )
+
+    dtype_group_map = {
+        "PeriodicTorsionPotential": {
+            "container": hoomd.md.dihedral.Harmonic,  # Should this be periodic, ask Josh
```

Comment:
As of Hoomd 3.8, `hoomd.md.dihedral.Harmonic` is deprecated and I think we should use `hoomd.md.dihedral.Periodic`  However, `hoomd.md.dihedral.Periodic` isn't available in versions < 3.8.  Do we need a hoomd version check somewhere?

Reply:
Note, I think we will only try to support HOOMD 3 for GMSO (I will have a version check somewhere)